python/taichi/lang/ast/ast_transformer.py

```diff
@@ -102,6 +103,26 @@ def build_Assign(ctx, node):
                                                is_static_assign)
         return None
 
+    @staticmethod
+    def build_assign_slice(ctx, node_target, values, is_static_assign):
+        target = ASTTransformer.build_Subscript(ctx, node_target, get_ref=True)
+        if isinstance(node_target.value.ptr, Matrix):
```

Comment:
Put `target = ASTTransformer.build_Subscript(ctx, node_target, get_ref=True)` in this scope so as to avoid adding "build_target=False" to `build_assign_basic`

Reply:
A follow up question is that `node_target.value.ptr` is not populated before calling `build_Subscript`

Reply:
You can probably fetch the variable with: 

![2022-07-12 10-48-48 的屏幕截图](https://user-images.githubusercontent.com/22334008/178397787-1915c4c0-18fa-4791-b2ae-17efad944b11.png)

It's gonna show the object's type:
![2022-07-12 10-49-04 的屏幕截图](https://user-images.githubusercontent.com/22334008/178397858-780c4cf0-6377-4b3b-b8a9-3a73abebfaa8.png)

Reply:
`node_target.value.id` is parsed from the AST:

![2022-07-12 10-50-35 的屏幕截图](https://user-images.githubusercontent.com/22334008/178398039-6b6fa62b-0071-4b2e-9078-eb9938ed8281.png)

Reply:
This sounds good. My only concern is that this approach won't work for expressions that subscripts are not applied to an identifier. For example:
```python3
@ti.kernel
def test_kernel():
    vertlist = ti.Matrix([[0., 0., 0.] for i in range(3)])
    p = ti.Vector([1, 2, 3])
    mat_dict = {
        'x': vertlist
    }
    mat_dict['x'][1, :] = p.transpose()
    print(mat_dict['x'])
```

Reply:
Good point! I guess we'll have to stick with the `build_target` solution then :p